gobblin-runtime/src/main/java/org/apache/gobblin/runtime/DatasetTaskSummary.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.gobblin.runtime;
+
+/**
+ * A class returned by {@link org.apache.gobblin.runtime.SafeDatasetCommit} to provide metrics for the dataset
+ * that can be reported as a single event in the commit phase.
+ */
+public class DatasetTaskSummary {
```

Comment:
from the perspective of those reading these metrics, it would be nice to know whether the count reflects what's committed or a failure.  maybe a boolean?

(I'm presuming in the case of `COMMIT_SUCCESSFUL_TASKS` that the two could be comingled)